app/src/main/java/org/mozilla/fenix/settings/studies/StudiesAdapter.kt

```diff
@@ -0,0 +1,162 @@
+
+import android.annotation.SuppressLint
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.annotation.StringRes
+import androidx.annotation.VisibleForTesting
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import com.google.android.material.button.MaterialButton
+import org.mozilla.experiments.nimbus.internal.EnrolledExperiment
+import org.mozilla.fenix.R
+import org.mozilla.fenix.settings.studies.CustomViewHolder.SectionViewHolder
+import org.mozilla.fenix.settings.studies.CustomViewHolder.StudyViewHolder
+
+private const val VIEW_HOLDER_TYPE_SECTION = 0
+private const val VIEW_HOLDER_TYPE_STUDY = 1
+
+/**
+ * An adapter for displaying studies items. This will display information related to the state of
+ * a study such as active. In addition, it will perform actions such as removing an study.
+ *
+ * @property studiesDelegate Delegate that will provides method for handling
+ * the studies actions items.
+ * @param studies The list of studies.
+ */
+@Suppress("LargeClass")
+class StudiesAdapter(
+    private val studiesDelegate: StudiesAdapterDelegate,
+    studies: List<EnrolledExperiment>,
+    @VisibleForTesting
+    internal val shouldSubmitOnInit: Boolean = true
+) : ListAdapter<Any, CustomViewHolder>(DifferCallback) {
+    /**
+     * Represents all the studies that will be distributed in multiple headers like
+     * active, and completed, this helps to have the data source of the items,
+     * displayed in the UI.
+     */
+    @VisibleForTesting
+    internal var studiesMap: MutableMap<String, EnrolledExperiment> =
+        studies.associateBy({ it.slug }, { it }).toMutableMap()
+
+    init {
+        if (shouldSubmitOnInit) {
+            submitList(createListWithSections(studies))
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomViewHolder {
+        return when (viewType) {
+            VIEW_HOLDER_TYPE_STUDY -> createStudiesViewHolder(parent)
+            VIEW_HOLDER_TYPE_SECTION -> createSectionViewHolder(parent)
+            else -> throw IllegalArgumentException("Unrecognized viewType")
+        }
+    }
+
+    private fun createSectionViewHolder(parent: ViewGroup): CustomViewHolder {
+        val context = parent.context
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.studies_section_item, parent, false)
+        val titleView = view.findViewById<TextView>(R.id.title)
+        val divider = view.findViewById<View>(R.id.divider)
+        return SectionViewHolder(view, titleView, divider)
+    }
+
+    private fun createStudiesViewHolder(parent: ViewGroup): StudyViewHolder {
+        val context = parent.context
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.study_item, parent, false)
```

Comment:
Nit: you can make this a little smaller and get rid of the `inflater` val
```suggestion
        val view = LayoutInflater.from(context).inflate(R.layout.study_item, parent, false)
```